### DISCUSSION TITLE
Correct Way to Bind JSON Columns

### DISCUSSION THREAD

**Discussion by cjmellor** [livewire/livewire#5033] [1. Help]
Hello 👋

This isn't so much a request for help, but more a discussion on if the way I have done it is correct or not.

At work, we have to make each Model translatable, so I have opted to use Spatie's [Laravel translatable](https://spatie.be/docs/laravel-translatable/v6) package. The package needs the columns that are to be translatable to be JSON columns in the database, and look something like this:

```json
{"en": "Golf Visitor Experience"}
```

The package uses accessors/mutators behind the scenes so I can still do `$product->name` and I'll get the result back as a string (based on my locale) but when trying to find the data to a field in a Component, it doesn't show as a string, but rather an **Object** which is to be expected.

I seem to have got around this by mounting a new property with the value:

```php
public string $productName;

public function mount()
{
    $this->productName = $this->product->name;
}
```

And now in the Blade file, I can bind to it:

```html
<input wire:model="productName" />
```

And when updating the field, I just have to assign the new property and save it:

```php
$this->product->name = $this->productName;

$this->product->save();
```

And yeah, this all works fine, but it took a while to figure out as there seems to be no information out there on this (that I could find) so I am just curious to know if this is a good way of working with JSON to bind to a value, or could it be done differently?

Thanks.

**Comment by joshhanley** [comment]
@cjmellor sometimes, I think the simple solution is the best one, assigning the data to a property and then saving it back to the model at the end, is probably the best way instead of trying to get Livewire to serialise JSON attributes properly.

Hope this helps!
